dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java

```diff
@@ -0,0 +1,240 @@
+import org.hisp.dhis.reporttable.ReportTable;
+
+/**
+ * This is just a helper class responsible for converting from/to Visualization.
+ * Such conversions are needed to keep backward compatibility across
+ * ReportTable, Chart and Visualization. It's just a temporary class an should
+ * be removed as soon as ReportTable and Chart are remove from the codebase.
+ *
+ * @author maikel arabori
+ */
+@Deprecated
+public class ConversionHelper
+{
+    public static List<Chart> convertToChartList( List<Visualization> entities )
+    {
+        final List<Chart> charts = new ArrayList<>();
+
+        if ( isNotEmpty( entities ) )
+        {
+            for ( final Visualization visualization : entities )
+            {
+                final Chart chart = new Chart();
+                copyProperties( visualization, chart, "type" );
+
+                // Consider only Visualization type that is a Chart
+                if ( visualization.getType() != null )
+                {
+                    // Set the correct type
+                    chart.setType( ChartType.valueOf( visualization.getType().name() ) );
+
+                    // Copy seriesItems
+                    if ( isNotEmpty( visualization.getOptionalAxes() ) )
+                    {
+                        final List<Series> seriesItems = new ArrayList<>();
+                        final List<Axis> axes = visualization.getOptionalAxes();
+
+                        for ( final Axis axis : axes )
+                        {
+                            final Series series = new Series();
+                            series.setSeries( axis.getDimensionalItem() );
+                            series.setAxis( axis.getAxis() );
+                            series.setId( axis.getId() );
+
+                            seriesItems.add( series );
+                        }
+                        chart.setSeriesItems( seriesItems );
+                    }
+
+                    // Copy column into series
+                    if ( isNotEmpty( visualization.getColumnDimensions() ) )
+                    {
+                        final List<String> columns = visualization.getColumnDimensions();
+                        chart.setSeries( columns.get( 0 ) );
+                    }
+
+                    // Copy rows into category
+                    if ( isNotEmpty( visualization.getRowDimensions() ) )
+                    {
+                        final List<String> rows = visualization.getRowDimensions();
+                        chart.setCategory( rows.get( 0 ) );
+                    }
+
+                    chart.setCumulativeValues( visualization.isCumulativeValues() );
+                    charts.add( chart );
+                }
+            }
+        }
+
+        return charts;
+    }
+
+    public static Visualization convertToVisualization( final Chart chart )
+    {
+        final Visualization visualization = new Visualization();
+
+        if ( chart != null )
```

Comment:
Maybe better with if chart == null then return? too many levels on indentation is not good